examplePrograms/library_components_export.py

```diff
@@ -0,0 +1,129 @@
+from nexarClients.design.nexarDesignClient import NexarClient
+import os
+import json
+import csv
+
+# This is the API query, feel free to chop and change this to suit your use case.
+# We recommend building queries in an IDE such as BananaCakePop or Postman.
+# You can use BananaCakePop in a browser here: https://api.nexar.com/graphql/
+
+workspaceURL = "Your workspace URL here"
+
+graphql_query = '''query library ($workspaceURL: String!, $after: String) {
+  desLibrary (workspaceUrl: $workspaceURL) {
+    components (first: 100, after: $after) {
+      totalCount
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      nodes {
+        id
+        folder {
+          name
+        }
+        description
+        name
+        comment
+        details {
+          parameters {
+            name
+            value
+          }
+        }
+        manufacturerParts {
+          partNumber
+          companyName
+        }
+      }
+    }
+  }
+}'''
+
+
+if __name__ == "__main__":
+
+    # You will need to set your Nexar application's client ID and secret in your environment variables.
+    # For this example, the application should have the design scope enabled.
+    # Alternatively, switch how these are inputted but be careful not to share your credentials.
+
+    client_id = os.environ["NEXAR_CLIENT_ID"]
+    client_secret = os.environ["NEXAR_CLIENT_SECRET"]
+
+    # This class provides the necessary functionality to fetch access tokens and query the API.
+    # The scopes provided allow us to query Altium 365 design data.
+
+    nexar = NexarClient(client_id, client_secret, ["design.domain", "openid", "user.access"])
+
+    def retrieve_components():
+
+        page_info = {"hasNextPage": True, "endCursor": None}
+        components = []
+
+        while page_info["hasNextPage"] is True:
```

Comment:
A simpler way to write this is `while page_info["hasNextPage"]:`.

If the value of `page_info["hasNextPage"] is True` is `True`, then of course the value of `page_info["hasNextPage"]` is `True` 🙂